### ISSUE TITLE
Synced data overwritten to file size of 0 bytes

### ISSUE THREAD

**Issue by audreytoskin** [owncloud/client#1218]
### Expected behaviour

Data should be synced between the server and client devices, with the full content of all files preserved.
### Actual behaviour

All the data in one folder have been overwritten to a file size of 0 bytes. The problem directory is now essentially just a list of empty file names. This has synced between my desktop and the server, and the files are not retrievable through the ownCloud trash bin or file revision histories, nor the local trash bin (since they weren't "deleted" per se?).
### Steps to reproduce

I honestly have no idea. This is the first time this has happened to me, and I'm trying _not_ to reproduce the error, because I've already lost hundreds of files and many, many hours of work. I could cry right now. 

As soon as I noticed the problem, I stopped the sync client and unmounted the hard drive until I can figure out how to recover the data. I don't know if this may be related, but I noticed the desktop client was also doing the thing where it re-downloaded or re-uploaded files which were already synced, which I see is already reported [here](https://github.com/owncloud/mirall/issues/994). I hadn't gotten around to trying to fix that before the present issue came up.
### Server configuration

Operating system: Debian Linux, kernel 2.6.32.45-grsec-2.2.2-r3

Web server: Apache httpd

Database: MySQL 5.1.56

PHP version: 5.4.20

ownCloud server version: 5.0.12

Storage backend: ...Um, the standard backend?
### Client configuration

Client version: Mirall 1.4.2

Operating system: Fedora Linux 19, 64-bit, kernel 3.11.8

OS language: English, US

Installation path of client: /usr/bin/owncloud
### Logs

**Output of `owncloud --logwindow`** pasted into a [gist](https://gist.github.com/Terrycloth/472a9f813c1c3531ab80)

I tried this, but I haven't been running the ownCloud client in logging mode, so this only had any logs for the time when I checked _after_ the 0-byte-size overwrite had already happened, so I don't know how useful this will be...

**Web server error log:**

Logs for yesterday, [November 27](https://gist.github.com/Terrycloth/99a60ee2611e2b4239fd)
Logs for the day before, [November 26](https://gist.github.com/Terrycloth/aaf6b214ea50ddbccae5)

**ownCloud log (data/owncloud.log):**

Read it [here](https://gist.github.com/Terrycloth/14776ebeb54882240aa0).

**Comment by hkernbach** [comment]
Also experienced this issue. Server Version 7.0.1. File 0 bytes @Client and @WebApplication

**Comment by jvillaraco** [comment]
It's still happening in Server version 8.0.2.

**Comment by syntaxys** [comment]
I have the same problem in 8.0.2 and could reproduce it by this way:
> mounting the files folder with webdavfs for importing local files on the machine
> copied a 700 MB file to the mount, it shows the accurate file size in the webdav mount!
> checked the file size via web frontend in owncloud: zero bytes!
> cd owncloud/path/to/files/copiedFile: zero bytes!
With some other large files but less than 700 MB I had no problems.

**Comment by butonic** [comment]
can you try with this patch (against 8.0.5):
```diff
diff --git a/lib/private/connector/sabre/file.php b/lib/private/connector/sabre/file.php
index e57d04f..8337c5a 100644
--- a/lib/private/connector/sabre/file.php
+++ b/lib/private/connector/sabre/file.php
@@ -175,7 +175,11 @@ class OC_Connector_Sabre_File extends OC_Connector_Sabre_Node implements \Sabre\
                        throw new \Sabre\DAV\Exception\ServiceUnavailable("Encryption is disabled");
                } else {
                        try {
-                               return $this->fileView->fopen(ltrim($this->path, '/'), 'rb');
+                               $res = $this->fileView->fopen(ltrim($this->path, '/'), 'rb');
+                               if ($res === false) {
+                                       throw new \Sabre\DAV\Exception\ServiceUnavailable("Could not open file");
+                               }
+                               return $res;
                        } catch (\OCA\Files_Encryption\Exception\EncryptionException $e) {
                                throw new \Sabre\DAV\Exception\Forbidden($e->getMessage());
                        } catch (\OCP\Files\StorageNotAvailableException $e) {
```

**Comment by angelus2014** [comment]
Noticed the same thing here. Strange thing is that when I put a backup of the file in my local ownCloud folder, it doesn't get synced to the server

**Comment by dragotin** [comment]
@angelus2014 make sure to change the modification time, it might be possible that your backup restore preserves that.

**Comment by angelus2014** [comment]
I can't change every file that is now 0 bytes on the server.... there are 100's as far as I can tell. I synced the files with rsync to the files directory, change the owner/group to www-data and now I have permission problem through the web-ui as well as through the sync application. So I thought, let's make a new user and put the stuff there. I can't even make a new user, claiming that the user already exists..........

**Comment by horuga23** [comment]
Same behaviour here (Server 8.0.3). This is absolutly bad, and in this state this is not a piece of software one can work with in an production environment.
When you guys have bugfixed this, maybe sometime it is stable enough again. But who will test this against sensible data? When I have to make backups everywhere, what should be the use of owncloud?

**Comment by tigusoft** [comment]
We had similar problem.

Possible cause could be that server was out of free disk space (for at least a day). 
Did anyone else checked that? Do you know if you had free disk space left all the time?

In our case it zero'ed many files, including files that probably were not written to since months - not new files.

This is very serious issue. Please reopen it. @lmaestro

**Comment by tigusoft** [comment]
More details: server was 8.1.3 (stable), and before that we had one of the stable versions too.
The selected update channel is: stable.

Server is running Debian 7.
All files are on one partition /, and it was out of disk space for some time.

**Comment by angelus2014** [comment]
In my case the server was _not_ out of diskspace. But in defence of ownCloud, being out of diskspace will always lead to trouble, you can't blame the software for that.

**Comment by tigusoft** [comment]
@angelus2014 I would say program should detect this (check all disk operations for error) and go into maintenance mode until admin resolves this to minimize damage to other files.

**Comment by mikethrussell** [comment]
Hi guys, I'm still having this problem where local files are being synced as 0 bytes, despite the server/web app displaying a different filesize. Is this issue being worked on?

**Comment by angelus1969** [comment]
@mikethrussell apparently not, they even closed the issue as I can see. Good thing then that I moved away from owncloud, as I cannot trust it with important data if they treat serious problems like this.

**Comment by fleish** [comment]
I am also encountering this issue currently in 7.0.5. Since this issue is closed I guess I will open a new one.

**Comment by Dr4K4n** [comment]
Running 9.0.4 stable and experienced this problem today, caused by an owncloud-client (2.2.2-18.4) that hadn't synced for some weeks or so.

**Comment by mvysny** [comment]
REOPEN. I lost hundreds of photos and couple of hours trying to figure out what the hell is going on. Both newest OwnCloud client (2.2.4) and server is 9.1.1-1.2. This is definitely not OK. I waved the photos goodbye, only to find out that the problem manifests today as well. Fortunately other owncloud clients still have valid copies of photos, yet there is nothing I can do:
1. I cannot delete those 0-sized files: the delete op will propagate to all clients and I will loose the photos.
2. I cannot force owncloud to re-download the photo somehow, can I?

The problem in the client is that it spits out hundreds of errors that "server said that the file is 4 mb long but downloaded 0 bytes", and happily downloads 0-sized file, eventually uploading it to the server when you mass-move your files elsewhere. The file sits on client's FS for some time until you realize that your data is fucking gone. This is the worst of data corruption - silently eating your data. The most evil of viruses of old did this. This is absolutely highest priority to fix.

Now just went to the server and realized that the main problem lies there, see screenshot. Apparently there are TWO COPIES of files, one a correct one, second one zero sized.
![screenshot from 2016-10-05 08-57-20](https://cloud.githubusercontent.com/assets/254091/19102570/f07dce62-8ad9-11e6-87da-c352a6091c62.png)

**Comment by mvysny** [comment]
Workaround for me: find any zero-sized files with "find -size 0" and stop using owncloud until this is resolved.

**Comment by kaplandani** [comment]
amazing that this bug is still there, I've suggested  several times before to add functionality the will protect local folder from mass deletion (in this case mass corruption), and always answered that this is not the purpose of this software...
I suggesting again - have new settings: 
X Allow local deleted (will prevent the client from deleting local files at all)
X Allow Mass delete (will stop the client ask user permission when more than 3 files are being deleted or changed at the same time)
This feature will ensure users that the server can't harm local files without their consent. 
and this you can turn this off, it will remain the same for any users that don't want to be notified on this events....

**Comment by mourtzouchos** [comment]
Hi guys, since 2013 that I faced this serious problem I stopped using OwnCloud. Today I realize that the same problem is still here. I will setup a new Owncloud server and create test client installations in order to find out what happens. I aggree with @kaplandani that there have to be some security options in order to prevent mass corruption of files

**Comment by chrisgraham** [comment]
For those migrating from Dropbox and finding this issue, it may be caused by Dropbox's "Smart Sync" and nothing to do with ownCloud.
It seems Dropbox enabled Smart Sync for us and is set on a file-by-file (or possibly folder-by-folder) level on Dropbox's end. Even if you have it disabled in a local machine's config, it is working off the settings on Dropbox's end. It looks to the user like you have all your files synched onto disk, but actually on the underlying file system they are 0 bytes. If you then subsequently put the drive into a Linux machine to copy them off (e.g. mounting an HFS+ volume), Linux will not understand the filesystem-level mechanism for Smart Sync, and just copy them as 0 bytes.
The workaround is to explicitly set Smart Sync to 'Local' on the top level Dropbox folder and let it properly sync, before transferring the files onto ownCloud.

**Comment by olivuntu** [comment]
I'm experiencing the same issue syncing on a USB key with rsync launched from the graphical interface 'luckybackup' on linux. It seems to concern mostly new files created inbetween two syncs, but unsure about that. Very annoying anyway ! Has anyone out there any clue of what is going on ?

**Comment by houseofbards** [comment]
Hi everyone,
I was experiencing same problems with rsync and synchronization between two android devices; if I update a file in the source, I find the same file in the destination folder - but invariably at 0 bytes.
After several headaches I found a solution (for rsync, at least)!
I added --delay-updates (along with --delete) option. In this way, when rsync find a file to update, deletes the same file in the destination and (at the end of all the synchronization process) copy again the new file, with the correct number of bytes.
It worked like a charm, for me!

Hope this details give you a hint for your troubles ;-)
